front/app/modules/commercial/insights/services/insightsCategoriesSuggestionsTasks.ts

```diff
@@ -57,11 +62,37 @@ export async function insightsTriggerCategoriesSuggestionsTasks(
     }
   );
 
-  streams.fetchAllWith({
-    partialApiEndpoint: [
-      `insights/views/${insightsViewId}/tasks/category_suggestions`,
-      `insights/views/${insightsViewId}/inputs`,
-    ],
+  // Refetch pending tasks at an interval
+  const subscription = pollingStream.subscribe(() => {
+    streams.fetchAllWith({
+      partialApiEndpoint: [
+        `insights/views/${insightsViewId}/tasks/category_suggestions`,
+      ],
+    });
   });
+
+  insightsCategoriesSuggestionsTasksStream(insightsViewId, {
+    queryParameters: { categories, inputs },
+  })
+    .observable.pipe(
+      // Ignore the first emitted value coming from the hook
+      skip(1),
+      // Poll while there are pending tasks
+      takeWhile((response) => {
+        return response.data.length > 0;
+      }),
+      // Refetch inputs when there are no pending tasks
+      finalize(() => {
```

Comment:
Hey, @SaraBCL thanks for this! I am happy to trust your judgement about the load on the backend and add the input refetching back in in a subsequent pull request as I do agree it will improve usability!